### ISSUE TITLE
dataset

### ISSUE THREAD

**Issue by wanghuahua2019** [GeneZC/ASGCN#14]
你好，我想请问一下dataset里的数据是怎么转换为.raw的？有点搞不太懂唉

**Comment by GeneZC** [comment]
建议参考 https://github.com/howardhsu/ABSA_preprocessing

**Comment by azaitangqiao** [comment]
graph，tree文件是怎么得来的

**Comment by GeneZC** [comment]
@azaitangqiao plz refer to dependency_graph.py and dependency_tree.py.

**Comment by azaitangqiao** [comment]
thanks you

**Comment by wanghuahua2019** [comment]
谢谢啦

**Comment by azaitangqiao** [comment]
denom = torch.sum(adj, dim=2, keepdim=True) + 1
        output = torch.matmul(adj, hidden) / denom，这里对输出的值除以度数的原因是？

**Comment by GeneZC** [comment]
@azaitangqiao 做一个归一化

**Comment by azaitangqiao** [comment]
为什么呢，原因是什么，是因为output = torch.matmul(adj, hidden) 这里的output的值太大了吗？还是学习了Dot-product-attention里面除以一个根号下dk的操作。

**Comment by GeneZC** [comment]
@azaitangqiao 是的，原理类似。
